connectors/byoc.py

```diff
@@ -627,7 +630,12 @@ async def validate_filtering(self, validator):
                 if validation_result.state == FilteringValidationState.VALID:
                     filter_["active"] = filter_.get("draft")
 
-        await self.index.update(doc_id=self.id, doc={"filtering": filtering})
+        await self.index.update(
```

Comment:
I'm wondering if we could provide a method inside the client class itself handling sequence numbers. To me personally this feels more as a concern of the client and not the connector, but just thinking out loud. So we're having something like `update_with_concurrency_control` and the API looks the same as `update`, but uses concurrency control under the hood.

Reply:
>I'm wondering if we could provide a method inside the client class itself handling sequence numbers.

Did you mean the [ESIndex](https://github.com/elastic/connectors-python/blob/db83cac4a66e0687e3741da80475e6d0692ca738/connectors/es/index.py#L18) class?

> So we're having something like `update_with_concurrency_control` and the API looks the same as `update`, but uses concurrency control under the hood.

Could you elaborate?

Do you mean we should add this method to [ESIndex](https://github.com/elastic/connectors-python/blob/db83cac4a66e0687e3741da80475e6d0692ca738/connectors/es/index.py#L18) class?

```python
async def update_with_concurrency_control(self, es_document, doc):
    return await self.client.update(
        index=self.index_name,
        id=es_document.id,
        doc=doc,
        if_seq_no=es_document._seq_no,
        if_primary_term=es_document._primary_term,
    )
```

Reply:
Hi @timgrein , I will merge this PR first, we can make a separate PR if needed.